docs/src/man/subduction2D/subduction2D.md

```diff
@@ -0,0 +1,247 @@
+compute_ρg!(ρg[2], phase_ratios, rheology, args)
+compute_viscosity!(stokes, 1.0, phase_ratios, args, rheology, (-Inf, Inf))
+```
+where `(-Inf, Inf)` is the viscosity cutoff.
+
+## Boundary conditions
+```julia
+    flow_bcs     = FlowBoundaryConditions(;
+        free_slip = (left = true, right = true, top = true, bot = true),
+        no_slip   = (left = false, right = false, top = false, bot=false),
+    )
+    stokes.V.Vx .= PTArray([ x*εbg for x in xvi[1], _ in 1:ny+2])
+    stokes.V.Vy .= PTArray([-y*εbg for _ in 1:nx+2, y in xvi[2]])
+    flow_bcs!(stokes, flow_bcs) # apply boundary conditions
+    update_halo!(stokes.V.Vx, stokes.V.Vy)
+
+```
+
+## Pseuo-transient coefficients
+```julia
+pt_stokes   = PTStokesCoeffs(li, di; ϵ=1e-4,  CFL = 1 / √2.1)
+```
+
+## Just before solving the problem...
+In this benchmark we want to keep track of τII, the total time `ttot`, and the analytical elastic solution `sol`
+```julia
+ solution(ε, t, G, η) = 2 * ε * η * (1 - exp(-G * t / η))
+```
+and store their time history in the vectors:
+```julia
+    τII        = Float64[]
+    sol        = Float64[]
+    ttot       = Float64[]
+```
+
+## Advancing one time step
+
+1. Solve stokes
+```julia
+solve!(
+    stokes,
+    pt_stokes,
+    di,
+    flow_bcs,
+    ρg,
+    phase_ratios,
+    rheology,
+    args,
+    dt,
+    igg;
+    kwargs = (;
+        iterMax          = 150e3,
+        nout             = 200,
+        viscosity_cutoff = (-Inf, Inf),
+        verbose          = true
+    )
+)
+```
+2. calculate the second invariant and push to history vectors
+```julia
+tensor_invariant!(stokes.ε)
+push!(τII, maximum(stokes.τ.xx))
+
+@parallel (@idx ni .+ 1) multi_copy!(@tensor(stokes.τ_o), @tensor(stokes.τ))
+@parallel (@idx ni) multi_copy!(
+    @tensor_center(stokes.τ_o), @tensor_center(stokes.τ)
+)
+
+it += 1
+t  += dt
+
+push!(sol, solution(εbg, t, G0, η0))
+push!(ttot, t)
+```
+# Visualization
+We will use `Makie.jl` to visualize the results
+```julia
+using GLMakie
+```
+
+## Fields
+```julia
+ # visualisation of high density inclusion
+th    = 0:pi/50:3*pi;
+xunit = @. radius * cos(th) + 0.5;
+yunit = @. radius * sin(th) + 0.5;
+
+fig   = Figure(size = (1600, 1600), title = "t = $t")
+ax1   = Axis(fig[1,1], aspect = 1, title = L"\tau_{II}", titlesize=35)
+ax2   = Axis(fig[2,1], aspect = 1, title = L"E_{II}", titlesize=35)
+ax3   = Axis(fig[1,2], aspect = 1, title = L"\log_{10}(\varepsilon_{II})", titlesize=35)
+ax4   = Axis(fig[2,2], aspect = 1)
+heatmap!(ax1, xci..., Array(stokes.τ.II) , colormap=:batlow)
+heatmap!(ax2, xci..., Array(log10.(stokes.EII_pl)) , colormap=:batlow)
+heatmap!(ax3, xci..., Array(log10.(stokes.ε.II)) , colormap=:batlow)
+lines!(ax2, xunit, yunit, color = :black, linewidth = 5)
+lines!(ax4, ttot, τII, color = :black)
+lines!(ax4, ttot, sol, color = :red)
+hidexdecorations!(ax1)
+hidexdecorations!(ax3)
+save(joinpath(figdir, "$(it).png"), fig)
+fig
+```
+
+### Final model
+![](990.png)
```

Comment:
Did you upload the figures? This png link wont work for the docs

Reply:
I did not push indeed